### PR TITLE
OpenStack ref arch content for master

### PR DESCRIPTION
**Pull request by ahardin-rh**
@vikram-redhat This brings the content of https://github.com/openshift/openshift-docs/pull/12519 and its companion follow-up PR https://github.com/openshift/openshift-docs/pull/12540 into the master branch.

I will then get these changes cherry-picked into the 3.11 docs. 

In this PR, we need to ensure we address these points so that the latest changes are applied to both master and 3.11:
https://github.com/openshift/openshift-docs/pull/12519#issuecomment-430514159
https://github.com/openshift/openshift-docs/pull/12519#issuecomment-430619278

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@rlopez133 @shlao Is the content presented here all relevant to OCP 3.11? Please let me know if any changes are needed. Thanks!

**Comment by shlao** [comment]
@ahardin-rh  according to this line: "--enable="rhel-7-server-ose-3.10-rpms" 
It means that the content is  OCP-3.10

**Comment by rlopez133** [comment]
@ahardin-rh - Everything I wrote up initially was for 3.10 not 3.11. Doesn't mean most of it wouldn't apply but it was not written for the 3.11 use case.

**Comment by vikram-redhat** [comment]
@rlopez133 - do you have the bandwidth to review this content for validity for 3.11? If not, could someone else help out?

File: install_config/topics/ocp_osp_provisioning.adoc [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+----
+
+Within this inventory directory, the _all.yml_ file contains all the different
+parameters that must be set in to order to achieve successful provisioning of
+the RHOCP instances. The _OSEv3.yml_ file contains some references required by
+the _all.yml_ file plus all the different OpenShift cluster parameter installation
+customization one would want.
+
+[[all.yml]]
+==== all.yml
```
Comment (vikram-redhat) [inline install_config/topics/ocp_osp_provisioning.adoc]: @ahardin-rh - Will need to change this to something other than all.yml (for both the anchor and the title) as the customer portal doesn't build if left like this.

**Comment by rlopez133** [comment]
@vikram-redhat - I think once the doc is assumed ready we can test drive it. 3.10->3.11 . Just so you know there is a problem with 3.11 and Ansible 2.6. Basically stops the ability to use the OSP provisioner in the openshift-ansible. However, ShiftStack team is aware and trying to get it merged into the 3.11 openshift-ansible RPM

**Comment by ahardin-rh** [comment]
@rlopez133 @vikram-redhat I rebased again. I think the docs are ready for testing, but let me know if anything else is needed.

**Comment by vikram-redhat** [comment]
@rlopez133 -  bump. Could you review?

**Comment by rlopez133** [comment]
sorry I haven't got back, let me see if I can get someone in our grp to look at this.

**Comment by ahardin-rh** [comment]
@rlopez133 Thanks for your edits! I applied them and you can review via this preview build:
http://file.rdu.redhat.com/~ahardin/12052018/OpenStack-red-arch-master/install_config/configuring_openstack.html

I am looking at the formatting issues with the asterisks in this section:
http://file.rdu.redhat.com/~ahardin/12052018/OpenStack-red-arch-master/install_config/configuring_openstack.html#create-an-openstack-flavor

When I escape the formatting, the Travis build chokes. So, I will do my best to find a workaround. Let me know if there are any other issues. Thanks!

**Comment by rlopez133** [comment]
Things I noticed:

- Missing a "\" after "--enable rhel-7-server-openstack-13-rpms"
- "ansible > 2.5" should be "Requires Ansible 2.5"
- In the All YAML file, Optional (Recommended)... needs a # in front to make it a comment.
- That whole section included in the All YAML file has a weird color that doesn't match the rest of the text above it
- TODO: "Registering with Subscriptino Mangaer the OpenShift Container Instances" this needs to be changed (3.11 has code that finally handles this) but will likely be a completely separate PR as we discussed @ahardin-rh and since it doesn't break anything right now its not a big deal.

**Comment by ahardin-rh** [comment]
@rlopez133 I made the edits. Sorry for the delay!
Updated preview build: http://file.rdu.redhat.com/~ahardin/12172018/OpenStack-red-arch-master/install_config/configuring_openstack.html

I will get this merged in and, if we need more follow-up edits, we can open a separate PR. Thanks again!

**Pull request merged**